Week3/homework/3-step3.js

```diff
@@ -2,24 +2,36 @@
 
 // use a 'for' loop
 function repeatStringNumTimesWithFor(str, num) {
-  // add your code here
-  return str;
+  let repStr = " ";
+  for (let i = 0; i < num; i++) {
+    repStr += str;
+  }
+  return repStr;
 }
 
 console.log('for', repeatStringNumTimesWithFor('abc', 3));
 
 // use a 'while' loop
 function repeatStringNumTimesWithWhile(str, num) {
-  // add your code here
-  return str;
+  let repStr = " ";
+  while (num > 0) {
+    repStr += str;
+    num--;
+  }
+  return repStr;
 }
 
 console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 
 // use a 'do...while' loop
 function repeatStringNumTimesWithDoWhile(str, num) {
-  // add your code here
-  return str;
+  let repStr = " ";
+  let i = 0;
+  do {
+    repStr += str;
+    i++;
+  } while (i < num);
+  return repStr;
 }
 
-console.log('while', repeatStringNumTimesWithDoWhile('abc', 3));
+console.log('do while', repeatStringNumTimesWithDoWhile('abc', 3));
```

Comment:
Your do-while version fails to meet the test in codeFreeCamp: `repeatStringNumTimes("abc", -2)` should return `""`.

In fact, all your versions fail these tests because you initialised your `repStr` variable with a space rather than an empty string.